File: library/src/main/java/com/github/library/requestMethod/baseMethod.java
package com.github.library.requestMethod;

import com.github.library.RestClient;
import com.github.library.helper.LogHelper;

import java.util.concurrent.TimeUnit;


public abstract class baseMethod {

    LogHelper logHelper = new LogHelper(RestClient.class);

    protected static String calcTime(Long startTime) {
        Long duration = getTimeMillisecond() - startTime;
        return String.valueOf(TimeUnit.MILLISECONDS.toSeconds(duration)) + "." + String.valueOf(TimeUnit.MILLISECONDS.toMillis(duration) % 1000).substring(0, 1);
    }

    protected static Long getTimeMillisecond() {
        return System.currentTimeMillis();
    }



}

File: library/src/main/java/com/github/library/requestMethod/baseMethod.kt
package com.github.library.requestMethod


import com.github.library.RestClient
import com.github.library.helper.LogHelper

import java.util.concurrent.TimeUnit

abstract class BaseMethod {

    internal var logHelper = LogHelper(RestClient::class.java)

    companion object {

        protected fun calcTime(startTime: Long?): String {
            val duration = timeMillisecond - startTime!!
            return TimeUnit.MILLISECONDS.toSeconds(duration).toString() + "." + (TimeUnit.MILLISECONDS.toMillis(duration) % 1000).toString().substring(0, 1)
        }

        private val timeMillisecond: Long
            get() = System.currentTimeMillis()
    }

}
